src/python_testing/matter_testing_infrastructure/chip/testing/metadata.py

```diff
@@ -20,9 +20,11 @@
 import yaml
 
 
-def bool_from_str(value: str) -> bool:
+def cast_to_bool(value: Any) -> bool:
```

Comment:
why is this needed? would yaml not auto-convert "true/false" to a bool type rather than having everything string?

Reply:
ah, this is for the non-yaml case since we support both. I wonder if we should explain (and maybe eventually TODO to remove the old syntax and leave only one syntax available)

Reply:
I've added TODO item: https://github.com/project-chip/connectedhomeip/issues/35787